### PR TITLE
Don't reuse unprivileged cache when building as root

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
When using --incremental and first building an image unprivileged, followed by building it as root, we can't reuse the cache as the ownership will be wrong. So let's make sure we don't reuse the cache when that's the case.

### CODE REVIEW

File: mkosi/__init__.py [hunk 1]
```diff
@@ -1753,6 +1753,10 @@ def reuse_cache(state: MkosiState) -> bool:
     else:
         return False
 
+    for p in (final, build):
+        if p.exists() and p.stat().st_uid != 0:
```
Comment (behrmann) [inline mkosi/__init__.py:1757]: Wouldn't that stop reusing the cache whenever it is not owned by root, i.e. also when one only ever does unprivileged builds?

Comment (DaanDeMeyer) [reply mkosi/__init__.py:1757]: No because we're in the user namespace so it will appear as if owned by root.

Comment (behrmann) [reply mkosi/__init__.py:1757]: Right, sufficiently far down in the stack. A comment for posteriority would probably make sense.

**Review by behrmann — APPROVED** [review]

**Comment by DaanDeMeyer** [comment]
Just added a comment so let's merge

**Pull request merged**